discord/abc.py

```diff
@@ -845,6 +845,13 @@ async def send(self, content=None, *, tts=False, embed=None, file=None,
 
             .. versionadded:: 1.4
 
+        message_reference: :class:`~discord.MessageReference`
```

Comment:
I don't really like the design of this parameter and its typing. Having the user explicitly pass in a wrapped class is verbose. It's probably better to allow an implicit conversion so a user can just pass in a `Message` object as well.

```suggestion
        reference: Union[:class:`~discord.Message`, :class:`~discord.MessageReference`]
```

There is however, an issue with the way this is laid out. Currently either approach requires a full `Message` object, which might not be desirable. I think `MessageReference` should be reconstructed in such a way that it functions similar to `PartialEmoji` where the state parameter injected is optional and only done if within the library -- that way a user can construct their own `MessageReference` without relying on any sort of message caching at all and just operating on IDs. Right now the `state` is only used in a few scenarios and it needs to be guarded.